wp-content/themes/alliance-child/demo data/default/theme_options.txt
a:474:{s:17:"trx_addons_action";s:0:"";s:32:"trx_addons_activated_addons_list";a:9:{s:13:"audio-effects";b:1;s:9:"bg-canvas";b:1;s:13:"image-effects";b:1;s:12:"mouse-helper";b:1;s:15:"secondary-image";b:1;s:9:"bg-slides";b:1;s:10:"image-mask";b:1;s:12:"image-scroll";b:1;s:10:"bp-widgets";b:1;}s:24:"trx_addons_admin_message";s:0:"";s:30:"trx_addons_cpt_layouts_created";s:1:"1";s:25:"trx_addons_just_activated";s:2:"no";s:18:"trx_addons_options";a:70:{s:10:"debug_mode";i:0;s:28:"disable_widgets_block_editor";s:1:"1";s:12:"retina_ready";s:1:"1";s:14:"images_quality";s:2:"60";s:19:"animate_inner_links";i:0;s:16:"add_target_blank";i:0;s:12:"popup_engine";s:8:"magnific";s:14:"page_preloader";s:6:"square";s:23:"page_preloader_bg_color";s:7:"#ffffff";s:20:"page_preloader_image";s:0:"";s:15:"scroll_progress";s:4:"hide";s:13:"scroll_to_top";s:1:"1";s:15:"hide_fixed_rows";i:0;s:13:"smooth_scroll";i:0;s:13:"menu_collapse";s:1:"1";s:18:"menu_collapse_icon";s:29:"trx_addons_icon-ellipsis-vert";s:12:"menu_stretch";i:0;s:21:"breadcrumbs_max_level";s:1:"3";s:16:"search_for_terms";i:0;s:27:"optimize_css_and_js_loading";s:4:"full";s:13:"disable_emoji";s:1:"1";s:22:"move_scripts_to_footer";s:1:"1";s:30:"move_scripts_to_footer_exclude";s:0:"";s:18:"async_scripts_load";s:1:"1";s:21:"async_scripts_exclude";s:0:"";s:9:"lazy_load";s:1:"1";s:14:"lazy_load_page";s:1:"4";s:16:"lazy_load_single";s:1:"0";s:13:"cache_handler";s:5:"files";s:24:"disable_css_and_js_cache";i:0;s:19:"remove_ver_from_url";s:1:"1";s:10:"ajax_views";i:0;s:13:"layouts_cache";i:0;s:19:"layouts_cache_types";a:5:{s:6:"header";s:1:"1";s:6:"footer";s:1:"0";s:7:"sidebar";s:1:"0";s:4:"blog";s:1:"0";s:6:"custom";s:1:"0";}s:21:"layouts_cache_popular";s:1:"1";s:10:"menu_cache";i:0;s:15:"api_google_load";s:1:"1";s:10:"api_google";s:0:"";s:17:"api_google_marker";s:0:"";s:18:"api_google_cluster";s:0:"";s:20:"api_google_analitics";s:0:"";s:22:"api_google_remarketing";s:0:"";s:13:"api_fb_app_id";s:0:"";s:14:"login_via_ajax";s:1:"1";s:17:"login_via_socials";s:0:"";s:29:"notify_about_new_registration";s:2:"no";s:26:"double_opt_in_registration";s:1:"1";s:7:"socials";a:4:{i:0;a:4:{s:4:"name";s:13:"icon-facebook";s:5:"title";s:8:"Facebook";s:3:"url";s:39:"https://www.facebook.com/ThemeRexStudio";s:5:"color";s:0:"";}i:1;a:4:{s:4:"name";s:12:"icon-twitter";s:5:"title";s:7:"Twitter";s:3:"url";s:34:"https://twitter.com/ThemerexThemes";s:5:"color";s:0:"";}i:2;a:4:{s:4:"name";s:15:"icon-instagramm";s:5:"title";s:10:"Instagramm";s:3:"url";s:39:"https://www.instagram.com/themerex_net/";s:5:"color";s:0:"";}i:3;a:4:{s:4:"name";s:12:"icon-youtube";s:5:"title";s:7:"Youtube";s:3:"url";s:56:"https://www.youtube.com/channel/UCnFisBimrK2aIE-hnY70kCA";s:5:"color";s:0:"";}}s:5:"share";a:4:{i:0;a:4:{s:4:"name";s:12:"icon-twitter";s:5:"title";s:0:"";s:3:"url";s:50:"//twitter.com/intent/tweet?text={title}&url={link}";s:5:"color";s:0:"";}i:1;a:4:{s:4:"name";s:13:"icon-facebook";s:5:"title";s:0:"";s:3:"url";s:45:"//www.facebook.com/sharer/sharer.php?u={link}";s:5:"color";s:0:"";}i:2;a:4:{s:4:"name";s:9:"icon-mail";s:5:"title";s:4:"Mail";s:3:"url";s:85:"//connect.mail.ru/share?url={link}&title={title}&description={descr}&imageurl={image}";s:5:"color";s:0:"";}i:3;a:4:{s:4:"name";s:9:"icon-link";s:5:"title";s:4:"Link";s:3:"url";s:142:"json:{"link": "#","attributes": {"title": "Copy URL to clipboard","data-message": "Copied!","data-copy-link-url": "{link}","nopopup": "true"}}";s:5:"color";s:0:"";}}s:11:"add_og_tags";s:1:"1";s:16:"emotions_allowed";s:1:"0";s:8:"emotions";a:1:{i:0;a:2:{s:4:"name";s:0:"";s:5:"title";s:0:"";}}s:13:"posts_banners";s:5:"Array";s:16:"scroll_to_anchor";s:1:"0";s:27:"update_location_from_anchor";s:1:"0";s:11:"input_hover";s:7:"default";s:11:"sc_ids_type";s:8:"advanced";s:19:"use_ajax_to_get_ids";s:1:"1";s:15:"sc_tabs_layouts";s:1:"0";s:14:"layouts_create";s:0:"";s:18:"columns_wrap_class";s:0:"";s:24:"columns_wrap_class_fluid";s:0:"";s:12:"column_class";s:0:"";s:15:"page_wrap_class";s:18:".page_content_wrap";s:11:"demo_enable";i:0;s:12:"demo_referer";s:0:"";s:16:"demo_default_url";s:0:"";s:9:"demo_logo";s:0:"";s:14:"demo_post_type";s:0:"";s:13:"demo_taxonomy";s:0:"";}s:21:"trx_addons_permalinks";a:42:{s:19:"cars_post_type_slug";s:0:"";s:23:"cars_taxonomy_type_slug";s:0:"";s:25:"cars_taxonomy_status_slug";s:0:"";s:24:"cars_taxonomy_maker_slug";s:0:"";s:24:"cars_taxonomy_model_slug";s:0:"";s:27:"cars_taxonomy_features_slug";s:0:"";s:25:"cars_taxonomy_labels_slug";s:0:"";s:23:"cars_taxonomy_city_slug";s:0:"";s:26:"cars_agents_post_type_slug";s:0:"";s:25:"cars_agents_taxonomy_slug";s:0:"";s:27:"certificates_post_type_slug";s:0:"";s:22:"courses_post_type_slug";s:0:"";s:21:"courses_taxonomy_slug";s:0:"";s:21:"dishes_post_type_slug";s:0:"";s:20:"dishes_taxonomy_slug";s:0:"";s:22:"layouts_post_type_slug";s:0:"";s:21:"layouts_taxonomy_slug";s:0:"";s:24:"portfolio_post_type_slug";s:0:"";s:23:"portfolio_taxonomy_slug";s:0:"";s:25:"properties_post_type_slug";s:0:"";s:29:"properties_taxonomy_type_slug";s:0:"";s:31:"properties_taxonomy_status_slug";s:0:"";s:33:"properties_taxonomy_features_slug";s:0:"";s:31:"properties_taxonomy_labels_slug";s:0:"";s:32:"properties_taxonomy_country_slug";s:0:"";s:30:"properties_taxonomy_state_slug";s:0:"";s:29:"properties_taxonomy_city_slug";s:0:"";s:37:"properties_taxonomy_neighborhood_slug";s:0:"";s:21:"agents_post_type_slug";s:0:"";s:20:"agents_taxonomy_slug";s:0:"";s:21:"resume_post_type_slug";s:0:"";s:23:"services_post_type_slug";s:0:"";s:22:"services_taxonomy_slug";s:0:"";s:27:"competitions_post_type_slug";s:0:"";s:26:"competitions_taxonomy_slug";s:0:"";s:21:"rounds_post_type_slug";s:0:"";s:22:"matches_post_type_slug";s:0:"";s:22:"players_post_type_slug";s:0:"";s:19:"team_post_type_slug";s:0:"";s:18:"team_taxonomy_slug";s:0:"";s:27:"testimonials_post_type_slug";s:0:"";s:26:"testimonials_taxonomy_slug";s:0:"";}s:31:"trx_addons_roles_and_caps_added";s:1:"1";s:34:"trx_addons_setup_elementor_options";s:1:"1";s:35:"trx_addons_theme_alliance_activated";s:1:"1";s:34:"trx_addons_theme_basekit_activated";s:1:"1";s:18:"trx_addons_version";s:6:"2.18.1";s:24:"trx_addons_widgets_areas";a:1:{i:0;a:3:{s:2:"id";i:1;s:4:"name";s:16:"Widgets Panel #1";s:11:"description";s:0:"";}}s:8:"blogname";s:8:"Alliance";s:15:"blogdescription";s:39:"Intranet &amp; Extranet WordPress Theme";s:9:"site_icon";s:4:"8564";s:14:"posts_per_page";s:1:"8";s:13:"show_on_front";s:4:"page";s:13:"page_on_front";s:4:"3665";s:14:"page_for_posts";s:3:"111";s:12:"sticky_posts";a:1:{i:0;i:349;}s:26:"wp_page_for_privacy_policy";s:1:"0";s:29:"rtmedia_activity_done_upgrade";s:1:"1";s:26:"rtmedia_current_media_slug";s:5:"media";s:34:"rtmedia_flush_rules_plugin_version";s:6:"4.6.13";s:34:"rtmedia_inspirebook_release_notice";s:4:"hide";s:38:"rtmedia_media_activity_upgrade_pending";s:1:"0";s:39:"rtmedia_media_size_import_pending_count";s:1:"0";s:28:"rtmedia_premium_addon_notice";s:4:"hide";s:21:"rtmedia-global-albums";a:1:{i:0;i:1;}s:15:"rtmedia-options";a:67:{s:22:"general_enableComments";s:1:"1";s:19:"general_enableLikes";s:1:"1";s:27:"general_enableGallerysearch";s:1:"1";s:22:"general_enableLightbox";s:1:"1";s:20:"general_perPageMedia";d:4;s:21:"general_display_media";s:9:"load_more";s:26:"buddypress_enableOnProfile";s:1:"1";s:24:"buddypress_enableOnGroup";s:1:"1";s:27:"buddypress_enableOnActivity";s:1:"1";s:26:"buddypress_limitOnActivity";s:1:"0";s:29:"buddypress_enableNotification";s:1:"1";s:28:"buddypress_mediaLikeActivity";s:1:"1";s:31:"buddypress_mediaCommentActivity";s:1:"1";s:26:"buddypress_enableOnComment";s:1:"1";s:40:"rtmedia_disable_media_in_commented_media";s:1:"1";s:20:"general_enableAlbums";s:1:"1";s:32:"general_enableAlbums_description";s:1:"1";s:26:"allowedTypes_photo_enabled";s:1:"1";s:26:"allowedTypes_video_enabled";s:1:"1";s:26:"allowedTypes_music_enabled";s:1:"1";s:34:"defaultSizes_photo_thumbnail_width";s:3:"150";s:35:"defaultSizes_photo_thumbnail_height";s:3:"150";s:33:"defaultSizes_photo_thumbnail_crop";s:1:"1";s:31:"defaultSizes_photo_medium_width";s:1:"0";s:32:"defaultSizes_photo_medium_height";s:1:"0";s:30:"defaultSizes_photo_large_width";s:1:"0";s:31:"defaultSizes_photo_large_height";s:1:"0";s:39:"defaultSizes_video_activityPlayer_width";s:1:"0";s:40:"defaultSizes_video_activityPlayer_height";s:1:"0";s:37:"defaultSizes_video_singlePlayer_width";s:1:"0";s:38:"defaultSizes_video_singlePlayer_height";s:1:"0";s:39:"defaultSizes_music_activityPlayer_width";s:1:"0";s:37:"defaultSizes_music_singlePlayer_width";s:1:"0";s:35:"defaultSizes_featured_default_width";s:3:"100";s:36:"defaultSizes_featured_default_height";s:3:"100";s:34:"defaultSizes_featured_default_crop";s:1:"1";s:26:"general_jpeg_image_quality";s:2:"90";s:15:"privacy_default";s:1:"0";s:14:"styles_enabled";s:1:"1";s:13:"styles_custom";s:0:"";s:30:"general_upload_terms_page_link";s:2:"#d";s:28:"general_upload_terms_message";s:18:"terms of services.";s:34:"general_upload_terms_error_message";s:30:"Please check terms of service.";s:36:"general_upload_terms_privacy_message";s:50:"Show "Terms of Service" checkbox on upload screen ";s:22:"general_downloadButton";s:1:"0";s:27:"general_enableMediaEndPoint";s:1:"0";s:21:"general_showAdminMenu";s:1:"0";s:19:"general_videothumbs";i:2;s:23:"general_uniqueviewcount";s:1:"0";s:17:"general_viewcount";s:1:"0";s:21:"general_AllowUserData";s:1:"0";s:20:"rtmedia_add_linkback";s:1:"0";s:20:"rtmedia_affiliate_id";s:1:"0";s:18:"rtmedia_enable_api";s:1:"0";s:22:"general_masonry_layout";s:1:"0";s:31:"general_masonry_layout_activity";s:1:"0";s:21:"general_direct_upload";s:1:"0";s:27:"allowedTypes_photo_featured";s:1:"0";s:27:"allowedTypes_video_featured";s:1:"0";s:27:"allowedTypes_music_featured";s:1:"0";s:30:"defaultSizes_photo_medium_crop";s:1:"0";s:29:"defaultSizes_photo_large_crop";s:1:"0";s:15:"privacy_enabled";s:1:"0";s:20:"privacy_userOverride";s:1:"0";s:27:"general_enable_upload_terms";s:1:"0";s:41:"general_upload_terms_show_pricacy_message";s:1:"0";s:28:"activity_enable_upload_terms";s:1:"0";}s:37:"rtmedia-update-template-notice-v3_9_4";s:4:"show";s:42:"widget_bp_messages_sitewide_notices_widget";a:2:{i:1;a:0:{}s:12:"_multiwidget";i:1;}s:36:"widget_trx_addons_widget_bp_messages";a:1:{s:12:"_multiwidget";i:1;}s:50:"widget_trx_addons_widget_bp_messages_notifications";a:1:{s:12:"_multiwidget";i:1;}s:23:"bp-better-chat-settings";a:175:{s:9:"mechanism";s:4:"ajax";s:8:"template";s:8:"standard";s:15:"thread_interval";s:1:"3";s:13:"site_interval";s:2:"10";s:15:"messagesPerPage";s:2:"20";s:18:"attachmentsFormats";a:0:{}s:20:"attachmentsRetention";s:3:"365";s:17:"attachmentsEnable";s:1:"0";s:15:"attachmentsHide";s:1:"1";s:21:"attachmentsRandomName";s:1:"1";s:18:"attachmentsMaxSize";s:2:"64";s:15:"miniChatsEnable";s:1:"0";s:14:"searchAllUsers";s:1:"0";s:14:"disableSubject";s:1:"0";s:20:"disableEnterForTouch";s:1:"1";s:16:"disableTapToOpen";s:1:"0";s:14:"autoFullScreen";s:1:"0";s:11:"mobilePopup";s:1:"0";s:16:"mobileFullScreen";s:1:"1";s:8:"chatPage";s:1:"0";s:14:"messagesStatus";s:1:"0";s:19:"allowDeleteMessages";s:1:"0";s:24:"disableDeleteThreadCheck";s:1:"0";s:9:"fastStart";s:1:"1";s:17:"miniThreadsEnable";s:1:"0";s:17:"miniFriendsEnable";s:1:"0";s:11:"friendsMode";s:1:"0";s:16:"singleThreadMode";s:1:"0";s:24:"redirectToExistingThread";s:1:"0";s:19:"disableGroupThreads";s:1:"0";s:20:"replaceStandardEmail";s:1:"1";s:12:"oEmbedEnable";s:1:"1";s:22:"disableEnterForDesktop";s:1:"0";s:14:"rateLimitReply";a:12:{s:6:"editor";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:6:"author";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:11:"contributor";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:10:"subscriber";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:21:"trx_addons_cars_agent";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:27:"trx_addons_properties_agent";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:10:"lp_teacher";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:8:"customer";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:12:"shop_manager";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:8:"employer";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:9:"candidate";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}s:12:"group_leader";a:2:{s:5:"value";s:1:"0";s:4:"type";s:4:"hour";}}s:21:"rateLimitReplyMessage";s:34:"Your limit for replies is exceeded";s:18:"restrictNewThreads";a:0:{}s:25:"restrictNewThreadsMessage";s:40:"You are not allowed to start new threads";s:20:"restrictBadWordsList";s:43:"Your message contains a word from blacklist";s:39:"restrictNewThreadsRemoveNewThreadButton";s:1:"0";s:18:"restrictNewReplies";a:0:{}s:25:"restrictNewRepliesMessage";s:28:"You are not allowed to reply";s:13:"restrictCalls";a:0:{}s:20:"restrictCallsMessage";s:34:"You are not allowed to make a call";s:20:"restrictViewMessages";a:0:{}s:27:"restrictViewMessagesMessage";s:14:"Message hidden";s:23:"restrictViewMiniThreads";a:0:{}s:23:"restrictViewMiniFriends";a:0:{}s:22:"restrictViewMiniGroups";a:0:{}s:19:"restrictMobilePopup";a:0:{}s:10:"videoCalls";s:1:"0";s:10:"audioCalls";

wp-content/plugins/trx_addons/addons/ai-helper/classes/Options.php
<?php
namespace TrxAddons\AiHelper;

if ( ! class_exists( 'Options' ) ) {

	/**
	 * Add options to the ThemeREX Addons Options
	 */
	class Options {

		/**
		 * Constructor
		 */
		function __construct() {
			add_filter( 'trx_addons_filter_options', array( $this, 'add_options' ) );
			add_filter( 'trx_addons_filter_before_show_options', array( $this, 'fix_options' ) );
			add_filter( 'trx_addons_filter_export_options', array( $this, 'remove_token_from_export' ) );
		}

		/**
		 * Add options to the ThemeREX Addons Options
		 * 
		 * @hooked trx_addons_filter_options
		 *
		 * @param array $options  Array of options
		 * 
		 * @return array  	  Modified array of options
		 */
		function add_options( $options ) {
			$is_options_page = trx_addons_get_value_gp( 'page' ) == 'trx_addons_options';
			$log_open_ai = $is_options_page ? Logger::instance()->get_log_report( 'open-ai') : '';
			$log_sd = $is_options_page ? Logger::instance()->get_log_report( 'stabble-diffusion') : '';
			trx_addons_array_insert_before( $options, 'sc_section', apply_filters( 'trx_addons_filter_options_ai_helper', array(
				'ai_helper_section' => array(
					"title" => esc_html__('AI Helper', 'trx_addons'),
					'icon' => 'trx_addons_icon-android',
					"type" => "section"
				),

				// Open AI API settings
				'ai_helper_info_openai' => array(
					"title" => esc_html__('Open AI', 'trx_addons'),
					"desc" => wp_kses_data( __("Settings of the AI Helper for Open AI API", 'trx_addons') )
							. ( ! empty( $log_open_ai ) ? wp_kses( $log_open_ai, 'trx_addons_kses_content' ) : '' ),
					"type" => "info"
				),
				'ai_helper_token_openai' => array(
					"title" => esc_html__('Open AI token', 'trx_addons'),
					"desc" => wp_kses( sprintf(
													__('Specify a token to use the OpenAI API. You can generate a token in your personal account using the link %s', 'trx_addons'),
													apply_filters( 'trx_addons_filter_openai_api_key_url',
																	'<a href="https://beta.openai.com/account/api-keys" target="_blank">https://beta.openai.com/account/api-keys</a>'
																)
												),
										'trx_addons_kses_content'
									),
					"std" => "",
					"type" => "text"
				),
				'ai_helper_model_openai' => array(
					"title" => esc_html__('Open AI model', 'trx_addons'),
					"desc" => wp_kses_data( __('Select a model to use with OpenAI API', 'trx_addons') ),
					"std" => "gpt-3.5-turbo",
					"options" => apply_filters( 'trx_addons_filter_ai_helper_list_models', Lists::get_list_ai_models(), 'openai' ),
					"type" => "select",
					"dependency" => array(
						"ai_helper_token_openai" => array('not_empty')
					),
				),
				'ai_helper_temperature_openai' => array(
					"title" => esc_html__('Temperature', 'trx_addons'),
					"desc" => wp_kses_data( __('Select a temperature to use with OpenAI API queries in the editor.', 'trx_addons') )
							. '<br />'
							. wp_kses_data( __('What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.', 'trx_addons') ),
					"std" => 1.0,
					"min" => 0,
					"max" => 2.0,
					"step" => 0.1,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_token_openai" => array('not_empty')
					),
				),

				// Stabble Diffusion API settings
				'ai_helper_info_stabble_diffusion' => array(
					"title" => esc_html__('Stabble Diffusion', 'trx_addons'),
					"desc" => wp_kses_data( __("Settings of the AI Helper for Stabble Diffusion API", 'trx_addons') )
							. ( ! empty( $log_sd ) ? wp_kses( $log_sd, 'trx_addons_kses_content' ) : '' ),
					"type" => "info"
				),
				'ai_helper_token_stabble_diffusion' => array(
					"title" => esc_html__('Stabble Diffusion token', 'trx_addons'),
					"desc" => wp_kses( sprintf(
													__('Specify a token to use the Stabble Diffusion API. You can generate a token in your personal account using the link %s', 'trx_addons'),
													apply_filters( 'trx_addons_filter_stabble_diffusion_api_key_url',
																	'<a href="https://stablediffusionapi.com/settings/api" target="_blank">https://stablediffusionapi.com/settings/api</a>'
																)
												),
										'trx_addons_kses_content'
									),
					"std" => "",
					"type" => "text"
				),
				'ai_helper_guidance_scale_stabble_diffusion' => array(
					"title" => esc_html__('Guidance scale', 'trx_addons'),
					"desc" => wp_kses_data( __('Scale for classifier-free guidance.', 'trx_addons') ),
					"std" => 7.5,
					"min" => 1,
					"max" => 20,
					"step" => 0.1,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_token_stabble_diffusion" => array('not_empty')
					),
				),
				'ai_helper_interference_steps_stabble_diffusion' => array(
					"title" => esc_html__('Interference steps', 'trx_addons'),
					"desc" => wp_kses_data( __('Number of denoising steps. Available values: 21, 31, 41, 51.', 'trx_addons') ),
					"std" => 21,
					"min" => 21,
					"max" => 51,
					"step" => 10,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_token_stabble_diffusion" => array('not_empty')
					),
				),
				'ai_helper_models_stabble_diffusion' => array(
					"title" => esc_html__("List of available models", 'trx_addons'),
					"desc" => wp_kses(
								sprintf(
									__("Specify id and name (title) for the each new model. A complete list of available models can be found at %s", 'trx_addons'),
									'<a href="https://stablediffusionapi.com/models" target="_blank">https://stablediffusionapi.com/models</a>'
								),
								'trx_addons_kses_content'
							),
					"dependency" => array(
						"ai_helper_token_stabble_diffusion" => array('not_empty')
					),
					"clone" => true,
					"std" => trx_addons_list_from_array( Lists::get_default_sd_models() ),
					"type" => "group",
					"fields" => array(
						"id" => array(
							"title" => esc_html__("Model ID", 'trx_addons'),
							"class" => "trx_addons_column-1_2",
							"std" => "",
							"type" => "text"
						),
						"title" => array(
							"title" => esc_html__("Title", 'trx_addons'),
							"class" => "trx_addons_column-1_2",
							"std" => "",
							"type" => "text"
						),
					)
				),

				// Image Generator
				'ai_helper_sc_igenerator_info' => array(
					"title" => esc_html__('Shortcode IGenerator', 'trx_addons'),
					"type" => "info"
				),
				'ai_helper_sc_igenerator_limits' => array(
					"title" => esc_html__('Use limits', 'trx_addons'),
					"desc" => wp_kses_data( __('Use limits (per hour and per visitor) when generating images.', 'trx_addons') ),
					"std" => "1",
					"type" => "switch"
				),
				'ai_helper_sc_igenerator_limit_per_hour' => array(
					"title" => esc_html__('Images per 1 hour', 'trx_addons'),
					"desc" => wp_kses_data( __('How many images can all visitors generate in 1 hour?', 'trx_addons') ),
					"std" => 12,
					"min" => 0,
					"max" => 1000,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_sc_igenerator_limits" => array(1)
					),
				),
				'ai_helper_sc_igenerator_limit_per_visitor' => array(
					"title" => esc_html__('Requests from 1 visitor', 'trx_addons'),
					"desc" => wp_kses_data( __('How many requests can send a single visitor in 1 hour?', 'trx_addons') ),
					"std" => 2,
					"min" => 0,
					"max" => 100,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_sc_igenerator_limits" => array(1)
					),
				),

				// Text Generator
				'ai_helper_sc_tgenerator_info' => array(
					"title" => esc_html__('Shortcode TGenerator', 'trx_addons'),
					"type" => "info"
				),
				'ai_helper_sc_tgenerator_temperature' => array(
					"title" => esc_html__('Temperature', 'trx_addons'),
					"desc" => wp_kses_data( __('What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.', 'trx_addons') ),
					"std" => 1,
					"min" => 0,
					"max" => 2,
					"step" => 0.1,
					"type" => "slider"
				),
				'ai_helper_sc_tgenerator_limits' => array(
					"title" => esc_html__('Use limits', 'trx_addons'),
					"desc" => wp_kses_data( __('Use limits (per request, per hour and per visitor) when generating text.', 'trx_addons') ),
					"std" => "1",
					"type" => "switch"
				),
				'ai_helper_sc_tgenerator_limit_per_request' => array(
					"title" => esc_html__('Max. tokens per 1 request', 'trx_addons'),
					"desc" => wp_kses_data( __('How many tokens can be used per one request to the API?', 'trx_addons') ),
					"std" => 1000,
					"min" => 0,
					"max" => 32000,
					"step" => 100,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_sc_tgenerator_limits" => array(1)
					),
				),
				'ai_helper_sc_tgenerator_limit_per_hour' => array(
					"title" => esc_html__('Requests per 1 hour', 'trx_addons'),
					"desc" => wp_kses_data( __('How many requests can be processed for all visitors in 1 hour?', 'trx_addons') ),
					"std" => 8,
					"min" => 0,
					"max" => 1000,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_sc_tgenerator_limits" => array(1)
					),
				),
				'ai_helper_sc_tgenerator_limit_per_visitor' => array(
					"title" => esc_html__('Requests from 1 visitor', 'trx_addons'),
					"desc" => wp_kses_data( __('How many requests can send a single visitor in 1 hour?', 'trx_addons') ),
					"std" => 2,
					"min" => 0,
					"max" => 100,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_sc_tgenerator_limits" => array(1)
					),
				),

				// Chat
				'ai_helper_sc_chat_info' => array(
					"title" => esc_html__('Shortcode AI Chat', 'trx_addons'),
					"type" => "info"
				),
				'ai_helper_sc_chat_temperature' => array(
					"title" => esc_html__('Temperature', 'trx_addons'),
					"desc" => wp_kses_data( __('What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.', 'trx_addons') ),
					"std" => 1,
					"min" => 0,
					"max" => 2,
					"step" => 0.1,
					"type" => "slider"
				),
				'ai_helper_sc_chat_limits' => array(
					"title" => esc_html__('Use limits', 'trx_addons'),
					"desc" => wp_kses_data( __('Use limits (per request, per hour and per visitor) when chatting.', 'trx_addons') ),
					"std" => "1",
					"type" => "switch"
				),
				'ai_helper_sc_chat_limit_per_request' => array(
					"title" => esc_html__('Max. tokens per 1 request', 'trx_addons'),
					"desc" => wp_kses_data( __('How many tokens can be used per one request to the chat?', 'trx_addons') ),
					"std" => 1000,
					"min" => 0,
					"max" => 32000,
					"step" => 100,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_sc_chat_limits" => array(1)
					),
				),
				'ai_helper_sc_chat_limit_per_hour' => array(
					"title" => esc_html__('Requests per 1 hour', 'trx_addons'),
					"desc" => wp_kses_data( __('How many requests can be processed for all visitors in 1 hour?', 'trx_addons') ),
					"std" => 80,
					"min" => 0,
					"max" => 1000,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_sc_chat_limits" => array(1)
					),
				),
				'ai_helper_sc_chat_limit_per_visitor' => array(
					"title" => esc_html__('Requests from 1 visitor', 'trx_addons'),
					"desc" => wp_kses_data( __('How many requests can send a single visitor in 1 hour?', 'trx_addons') ),
					"std" => 10,
					"min" => 0,
					"max" => 100,
					"type" => "slider",
					"dependency" => array(
						"ai_helper_sc_chat_limits" => array(1)
					),
				),
			) ) );

			return $options;
		}

		/**
		 * Fix option params in the ThemeREX Addons Options
		 * 
		 * @hooked trx_addons_filter_before_show_options
		 *
		 * @param array $options  Array of options
		 * 
		 * @return array  	  Modified array of options
		 */
		function fix_options( $options ) {
			$max_tokens = OpenAI::get_max_tokens();
			if ( ! empty( $options['ai_helper_sc_tgenerator_limit_per_request']['std'] ) && $options['ai_helper_sc_tgenerator_limit_per_request']['std'] > $max_tokens ) {
				$options['ai_helper_sc_tgenerator_limit_per_request']['std'] = $max_tokens;
			}
			if ( ! empty( $options['ai_helper_sc_tgenerator_limit_per_request']['val'] ) && $options['ai_helper_sc_tgenerator_limit_per_request']['val'] > $max_tokens ) {
				$options['ai_helper_sc_tgenerator_limit_per_request']['val'] = $max_tokens;
			}
			if ( ! empty( $options['ai_helper_sc_tgenerator_limit_per_request']['max'] ) ) {
				$options['ai_helper_sc_tgenerator_limit_per_request']['max'] = $max_tokens;
			}
			if ( ! empty( $options['ai_helper_sc_chat_limit_per_request']['std'] ) && $options['ai_helper_sc_chat_limit_per_request']['std'] > $max_tokens ) {
				$options['ai_helper_sc_chat_limit_per_request']['std'] = $max_tokens;
			}
			if ( ! empty( $options['ai_helper_sc_chat_limit_per_request']['val'] ) && $options['ai_helper_sc_chat_limit_per_request']['val'] > $max_tokens ) {
				$options['ai_helper_sc_chat_limit_per_request']['val'] = $max_tokens;
			}
			if ( ! empty( $options['ai_helper_sc_chat_limit_per_request']['max'] ) ) {
				$options['ai_helper_sc_chat_limit_per_request']['max'] = $max_tokens;
			}
			return $options;
		}

		/**
		 * Clear some addon specific options before export
		 * 
		 * @hooked trx_addons_filter_export_options
		 * 
		 * @param array $options  Array of options
		 * 
		 * @return array  	  Modified array of options
		 */
		 function remove_token_from_export( $options ) {
			if ( isset( $options['trx_addons_ai_helper_log'] ) ) {
				unset( $options['trx_addons_ai_helper_log'] );
			}
			if ( ! empty( $options['trx_addons_options']['ai_helper_token_openai'] ) ) {
				$options['trx_addons_options']['ai_helper_token_openai'] = '';
			}
			if ( ! empty( $options['trx_addons_options']['ai_helper_token_stabble_diffusion'] ) ) {
				$options['trx_addons_options']['ai_helper_token_stabble_diffusion'] = '';
			}
			return $options;
		}
	}
}
